### ISSUE TITLE
[BUG] npm 10.2.0 npm publish to github get 400 Bad request response

### ISSUE THREAD

**Issue by benq-tyreke-t-yang** [npm/cli#6901]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

I originally used `npm 10.1.0` and encountered an issue where system proxy environment variables were not being applied during `npm publish`. So, I upgraded to `npm 10.2.0`, but instead, I encountered a `400 Bad Request` response with no additional clues.

npm notice === Tarball Details === 
npm notice Publishing to https://npm.pkg.github.com/ with tag latest and default access
npm ERR! code E400
npm ERR! 400 Bad request - PUT https://npm.pkg.github.com/@foo%2fbar

P.S. Everything works fine when I use the docker image `node:16.10.0-buster` or just `node:20.8.0-bookworm` with `npm 10.1.0` and a properly configured npm proxy.

### Expected Behavior

`npm publish` to `npm.pkg.github.com` successfully

### Steps To Reproduce

First, set up the npm.pkg.github.com registry and token through .npmrc or other means.

```
$ docker run -it -v /path/about/package:/usr/app  -w /usr/app node:20.8.0-bookworm bash
root@03fdd178ece1:/usr/app# npm install -g npm@10.2.0
root@03fdd178ece1:/usr/app# npm -v
10.2.0
root@03fdd178ece1:/usr/app# npm install && npm run build && npm publish
```

### Environment

- npm: 10.2.0
- Node.js: 20.8.0
- OS Name: debian bookworm
- System Model Name: Linux kernel
- npm config: default

**Comment by wraithgar** [comment]
Please fill out the entire template when submitting an issue. Of particular importance here is the config, which will show how npm is trying to apply auth to different registries.  As such there is not enough info here to reproduce.

**Comment by hax** [comment]
I meet the same issue. It seems when provide proxy via env or npm config, the request with token will fail with 400.

**Comment by hax** [comment]
npm 9 is ok, it seems it's npm 10 bug. (I also tested yarn and cnpm, they are ok)

**Comment by saw-lau** [comment]
Confirming that I have the same issue as well behind a firewall - version 10.2.4 fails but npm 9.6.3 w/ Node 20.0.0 works with no problems.

**Comment by david-korecko** [comment]
Still doesn't seem to be resolved?

**Comment by benq-tyreke-t-yang** [comment]
> Still doesn't seem to be resolved?

Check this link https://github.com/npm/cli/issues/7024#issuecomment-1991244913,
This has been resolved in npm v10.5.0

**Comment by david-korecko** [comment]
> > Still doesn't seem to be resolved?
> 
> Check this link [#7024 (comment)](https://github.com/npm/cli/issues/7024#issuecomment-1991244913), This has been resolved in npm v10.5.0

I've seen that, however, using npm v10.5.0 still produces the same issue in my case:

`npm notice Publishing to https://npm.customregistry.com/ with tag latest and restricted access
npm ERR! code E400
npm ERR! 400 Bad Request - PUT https://npm.customregistry.com/@customscope%2fcustom-package-name
npm ERR! A complete log of this run can be found in: /home/runner/.npm/_logs/2024-05-14T09_48_23_815Z-debug-0.log
Error: Command failed: npm publish --no-workspaces`

Could this maybe be caused by the registry itself, is the issue not fixed (I've confirmed I'm on the fixed npm version 10.5.0), or is there another issue with similar symptoms?
